sdks/python/apache_beam/runners/portability/fn_api_runner_test.py

```diff
@@ -753,6 +753,81 @@ def create_pipeline(self):
         runner=fn_api_runner.FnApiRunner(bundle_repeat=3))
 
 
+class EventRecorder(object):
+  def __init__(self):
+    self.work_path = os.getcwd() + '/EventRecorder'
+    if os.path.exists(self.work_path):
+      self.cleanup()
+    os.mkdir(self.work_path)
+
+  def record(self, file_name):
+    file_path = os.path.join(self.work_path, str(file_name) + '.txt')
```

Comment:
I don't think a finalizer could be called twice unless it's registered twice. If a finalizer callback gets registered twice, then it's the user's choice and I don't think we can say it's not a expected behavior. Instead, how about adding try-catch when calling callbacks in case the situation like, trying to delete one deleted file?

Reply:
For https://docs.python.org/2/library/tempfile.html:
I tried to use tempfile.NamedTemporaryFile but it made ParDo fail to initialize:
```
Traceback (most recent call last):
  File "/usr/local/google/home/boyuanz/beam/sdks/python/apache_beam/runners/portability/fn_api_runner_test.py", line 820, in test_register_finalizations
    fn = beam.ParDo(FinalizableDoFn())
  File "/usr/local/google/home/boyuanz/beam/sdks/python/apache_beam/transforms/core.py", line 953, in __init__
    super(ParDo, self).__init__(fn, *args, **kwargs)
  File "/usr/local/google/home/boyuanz/beam/sdks/python/apache_beam/transforms/ptransform.py", line 688, in __init__
    self.fn = pickler.loads(pickler.dumps(self.fn))
  File "/usr/local/google/home/boyuanz/beam/sdks/python/apache_beam/internal/pickler.py", line 258, in loads
    return dill.loads(s)
  File "/usr/local/google/home/boyuanz/beam/sdks/python/.eggs/dill-0.2.9-py2.7.egg/dill/_dill.py", line 317, in loads
    return load(file, ignore)
  File "/usr/local/google/home/boyuanz/beam/sdks/python/.eggs/dill-0.2.9-py2.7.egg/dill/_dill.py", line 305, in load
    obj = pik.load()
  File "/usr/lib/python2.7/pickle.py", line 864, in load
    dispatch[key](self)
  File "/usr/lib/python2.7/pickle.py", line 1221, in load_build
    setstate = getattr(inst, "__setstate__", None)
  File "/usr/lib/python2.7/tempfile.py", line 401, in __getattr__
    file = self.__dict__['file']
KeyError: 'file'
```
Planing using tempfile to create a tmp dir for EventRecorder